pkg/collector/reconcile/opentelemetry.go

```diff
@@ -38,6 +41,17 @@ import (
 func Self(ctx context.Context, params manifests.Params) error {
 	changed := params.Instance
 
+	up := &collectorupgrade.VersionUpgrade{
+		Log:      params.Log,
+		Version:  version.Get(),
+		Client:   params.Client,
+		Recorder: record.NewFakeRecorder(collectorupgrade.RecordBufferSize),
```

Comment:
I believe we should have a real recorder through params at this point, we should use that here

Reply:
So: should I add the recorder to the Params struct to propagate it here? It makes sense. I'll do it.

Thanks for the suggestion!